libs/portals/admin/ids-admin/src/screens/Client/components/AddPermissions/AddPermissions.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState } from 'react'
+import { useFetcher, useParams } from 'react-router-dom'
+
+import { AuthAdminClientAllowedScope } from '@island.is/api/schema'
+import {
+  Box,
+  Button,
+  Checkbox,
+  Option,
+  Select,
+  Table as T,
+  Text,
+  useBreakpoint,
+} from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import { getTranslatedValue } from '@island.is/portals/core'
+import { Modal } from '@island.is/react/components'
+import { isDefined } from '@island.is/shared/utils'
+
+import { ShadowBox } from '../../../../components/ShadowBox/ShadowBox'
+import { m } from '../../../../lib/messages'
+import { IDSAdminPaths } from '../../../../lib/paths'
+import { AuthTenants } from '../../../Tenants/Tenants.loader'
+import { useClient } from '../../ClientContext'
+import {
+  GetAvailableScopesQuery,
+  useGetAvailableScopesLazyQuery,
+} from './AddPermissions.generated'
+
+import * as styles from './AddPermissions.css'
+
+type AuthAdminScopes = GetAvailableScopesQuery['authAdminScopes']['data'][0]['environments']
+
+interface AddPermissionsProps {
+  isVisible: boolean
+  onClose: () => void
+  onAdd: (permissions: AuthAdminClientAllowedScope[]) => void
+  addedScopes: AuthAdminClientAllowedScope[]
+  removedScopes: AuthAdminClientAllowedScope[]
+}
+
+const formatOption = (tenant: AuthTenants[0], locale: string) =>
+  ({
+    label: getTranslatedValue(tenant.defaultEnvironment.displayName, locale),
+    value: tenant.id,
+  } as Option)
+
+export const AddPermissions = ({
+  isVisible,
+  onClose,
+  onAdd,
+  addedScopes,
+}: AddPermissionsProps) => {
+  const { formatMessage, locale } = useLocale()
+  const { tenant } = useParams()
+  const { md } = useBreakpoint()
+  const {
+    selectedEnvironment: { environment, allowedScopes },
+  } = useClient()
+  const fetcher = useFetcher()
+  const [selectedTenant, setSelectedTenant] = useState<Option>()
+  const [availableTenants, setAvailableTenants] = useState<AuthTenants>([])
+  const [availableScopes, setAvailableScopes] = useState<AuthAdminScopes>([])
+
+  const [
+    getAvailableScopesQuery,
+    { data: { authAdminScopes } = { authAdminScopes: undefined }, loading },
+  ] = useGetAvailableScopesLazyQuery({
+    fetchPolicy: 'network-only',
+  })
+
+  const [selected, setSelected] = useState<
+    Map<string, AuthAdminClientAllowedScope>
+  >(new Map())
+
+  useEffect(() => {
+    if (fetcher.state === 'idle' && !fetcher.data) {
+      fetcher.load(IDSAdminPaths.IDSAdmin)
+    }
+  }, [fetcher])
+
+  useEffect(() => {
+    if (fetcher.data) {
+      // When fetcher has loaded data we filter available tenants for the selected environment
+      const availableTenants = (fetcher.data as AuthTenants).filter((tenant) =>
```

Comment:
nitnit: Maybe it is a little cleaner to store fetcher.data in a var below fetcher.

```
  const fetcher = useFetcher<AuthTenants | undefined>()
  const tenants = fetcher.data
```